### PR TITLE
Reslice mapper pd

### PR DESCRIPTION
**Pull request by sankhesh**
### Context

The vtkImageResliceMapper allows reslicing the volume with custom polydata. However, when switching between a vtkPlane and vtkPolyData as the slicing function, the mapper was failing to update graphics buffers. 

Also fixed a regression in the vtkImplicitPlaneWidget due to recent changes in handleEvents API of widgets.
### Results

Screenshot from 2023-05-26 12-47-05

Screenshot from 2023-05-26 12-47-21

### Changes

- [x] Documentation and TypeScript definitions were updated to match those changes

### PR and Code Checklist

- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing

- [x] This change adds or fixes unit tests 
- [x] Tested environment:
  - **vtk.js**: 
  - **OS**: 
  - **Browser**:

### CODE REVIEW

File: Sources/Rendering/Core/ImageResliceMapper/example/index.js [hunk 1]
```diff
@@ -98,6 +107,11 @@ function setImage(im) {
     0.5 * (bds[2] + bds[3]),
     0.5 * (bds[5] + bds[4])
   );
+  slicePolyDataSource.setCenter(
+    0.5 * (bds[0] + bds[1]),
+    0.5 * (bds[2] + bds[3]),
+    0.5 * (bds[5] + bds[4])
```
Comment (finetjul) [inline Sources/Rendering/Core/ImageResliceMapper/example/index.js:113]: `0.5 * (bds[4] + bds[5])` or `vtkBoundingBox.getCenter(bds)` or even better `im.getCenter()`

File: Sources/Rendering/Core/ImageResliceMapper/example/index.js [hunk 2]
```diff
@@ -108,6 +122,12 @@ function setImage(im) {
   planeState.onModified(() => {
     slicePlane.setOrigin(planeState.getOrigin());
     slicePlane.setNormal(planeState.getNormal());
+    slicePolyDataSource.setCenter(planeState.getOrigin());
+    slicePolyDataSource.setDirection(planeState.getNormal());
+    if (mapper.getSlicePolyData()) {
+      slicePolyDataSource.update();
+      mapper.setSlicePolyData(slicePolyDataSource.getOutputData());
```
Comment (finetjul) [inline Sources/Rendering/Core/ImageResliceMapper/example/index.js:129]: shouldn't this work with connections: `setInputConnection(1, slicePolyDataSource.getOutputPort())`

**Review by finetjul — APPROVED** [review]

**Review by finetjul — APPROVED** [review]

File: Sources/Rendering/Core/ImageResliceMapper/example/index.js [hunk 3]
```diff
@@ -189,6 +209,19 @@ for (let idx = 0; idx < slabTypes.length; ++idx) {
     renderWindow.render();
   };
 }
+const sliceFunction = document.querySelectorAll('.spd');
+for (let idx = 0; idx < sliceFunction.length; ++idx) {
+  const st = sliceFunction[idx];
+  st.onchange = (e) => {
+    if (e.target.value === 'plane') {
+      mapper.setSlicePolyData(null);
```
Comment (jadh4v) [inline Sources/Rendering/Core/ImageResliceMapper/example/index.js:217]: Looks like the mapper switches back to Plane slicing if the polydata is set to `null`. This kinda feels too subtle. Class design wise, would it be better to have an explicit flag or integer to set the modes to implicit function vs polydata slicing? or handle the type internally and only have a single setter like `setSlicingGeometry`?

**Review by jadh4v — APPROVED** [review]
One comment, but otherwise LGTM

File: Sources/Rendering/Core/ImageResliceMapper/example/index.js (hunk 3 above)
Comment (sankhesh) [reply Sources/Rendering/Core/ImageResliceMapper/example/index.js:217]: This PR doesn't change that behavior and that's documented here: https://github.com/Kitware/vtk-js/blob/master/Sources/Rendering/Core/ImageResliceMapper/index.d.ts#L206-L214

When designing the mapper, I considered adding a flag to switch between the two but figured preferring the polydata input should be fine for most cases.

File: Sources/Rendering/Core/ImageResliceMapper/example/index.js (hunk 2 above)
Comment (sankhesh) [reply Sources/Rendering/Core/ImageResliceMapper/example/index.js:129]: It doesn't work as is but yeah, that could prove useful.

**Pull request merged**